### DISCUSSION TITLE
mitmproxy puts entire URL into GET

### DISCUSSION THREAD

**Discussion by Tillerino** [mitmproxy/mitmproxy#6588] [Q&A] [answered]
I'm sure I'm doing something completely wrong here.

1) backend is a simple echo server running on localhost:8081: `npm install http-echo-server; PORT=8081 npx http-echo-server`.

2) a reverse-proxy to this backend on port 5001: `mitmdump -p 5001 --mode reverse:http://127.0.0.1:8081`

3) a simple request: `curl -x http://localhost:5001 http://example.com/hello`

I would have expected to see a request for `/hello` on the backend, but instead I see a request for `http://example.com/hello` (output of 1):

```
[server] event: listening (port: 8081)
[server] event: connection (socket#1)
[socket#1] event: resume
[socket#1] event: data
--> GET http://example.com/hello HTTP/1.1
--> Host: 127.0.0.1:8081
--> User-Agent: curl/8.4.0
--> Accept: */*
--> Proxy-Connection: Keep-Alive
```

This is very strange, since `GET` doesn't usually contain the schema or host. Whereas mitm logs something entirely different (output of 2):

```
[16:43:30.257] reverse proxy to http://127.0.0.1:8081 listening at *:5001.
[16:43:33.014][[::1]:40476] client connect
[16:43:33.015][[::1]:40476] server connect 127.0.0.1:8081
[::1]:40476: GET http://127.0.0.1:8081/hello
          << 200 OK 130b
[16:43:35.022][[::1]:40476] server disconnect 127.0.0.1:8081
[16:43:35.022][[::1]:40476] client disconnect
```

Now this is no quirk of the frontend or backend. I've tried Firefox as a frontend instead of curl, and I've used a different backend (which is where I started) - all with the same result. What's going wrong here? Am I missing a configuration?

**Comment by mhils** [comment, marked as answer]
You are using reverse proxy mode (which is typically invisible to the client), but you are also configuring a proxy on the client side. As a result, your client sends an absolute-form request (intended for a proxy) to mitmproxy, which forwards it unmodified to the target server. You can either:

1. Not use a proxy, i.e. run `curl http://localhost:5001/hello`
2. Run mitmproxy in regular mode, i.e. `mitmdump -p 5001`

If you want to run a regular proxy but change where the request is going, check out https://docs.mitmproxy.org/stable/overview-features/#map-remote.

**Reply by Tillerino** [reply]
Ah yes, I see. So what I wanted to do here was something along the lines of:

`mitmdump -p 5001 --map-remote "|~m GET|//.+/|//localhost:8081/"`

Thanks for the quick response!
